server/dataone_package.py

```diff
@@ -24,57 +29,95 @@ def create_resource_map(resmap_pid, eml_pid, file_pids):
+    :type eml_pid: str
     :return: The EML as as string of bytes
     :rtype: bytes
     """
 
-    logger.debug('Entered create_minimum_eml')
-    top = '<?xml version="1.0" encoding="UTF-8"?>'
-    namespace = '<eml:eml xmlns:eml="eml://ecoinformatics.org/eml-2.1.1" ' \
-                'xmlns:stmml="http://www.xml-cml.org/schema/stmml-1.1" xmlns:' \
-                'xsi="http://www.w3.org/2001/XMLSchema-instance" packageId="' \
-                'doi:10.18739/A20Q25" system="https://arcticdata.io" xsi:' \
-                'schemaLocation="eml://ecoinformatics.org/eml-2.1.1 eml.xsd">'
-
-    dataset = '<dataset>\n'
-    title = '<title>{0}</title>\n'.format(str(tale.get('title', '')))
-
-    individual_name = '<individualName>\n<surName>\n{0}\n</surName>\n</individualName>'.format(
-        str(user.get('lastName', '')))
-
-    creator = '<creator>\n{0}\n</creator>\n'.format(individual_name)
-    contact = '<contact>\n{0}\n</contact>\n'.format(individual_name)
-    dataset_close = '</dataset>\n'
-    eml_close = '</eml:eml>'
-
-    # Append the above xml together to form the EML document
-    xml = top+namespace+dataset+title+creator+contact+dataset_close+eml_close
-
+    # The surName in the EML record is taken by the user's last name. If
+    # it doesn't exist, try to use the first name.
+    lastName = user.get('lastName', 'None')
+    firstName = user.get('firstName', 'None')
+    ns = ET.Element('eml:eml')
+
+    ns.set('xmlns:eml', "eml://ecoinformatics.org/eml-2.1.1")
+    ns.set('xsi:schemaLocation', "eml://ecoinformatics.org/eml-2.1.1 eml.xsd")
+    ns.set('xmlns:stmml', "http://www.xml-cml.org/schema/stmml-1.1")
+    ns.set('xmlns:xsi', "http://www.w3.org/2001/XMLSchema-instance")
+    ns.set('scope', "system")
+    ns.set('system', "knb")
+    ns.set('packageId', eml_pid)
+
+    dataset = ET.SubElement(ns, 'dataset')
+    ET.SubElement(dataset, 'title').text = str(tale.get('title', ''))
+
+    creator = ET.SubElement(dataset, 'creator')
+    individual_name = ET.SubElement(creator, 'individualName')
+    ET.SubElement(individual_name, 'givenName').text = firstName
+    ET.SubElement(individual_name, 'surName').text = lastName
+
+    abstract = ET.SubElement(dataset, 'abstract')
+    description = get_tale_description(tale)
+    ET.SubElement(abstract, 'para').text = description if\
+        bool(len(description)) else 'None'
+
+    contact = ET.SubElement(dataset, 'contact')
+    ind_name = ET.SubElement(contact, 'individualName')
+    ET.SubElement(ind_name, 'givenName').text = firstName
+    ET.SubElement(ind_name, 'surName').text = lastName
+
+    # Add a <otherEntity> block for each object
+    for item_id in item_ids:
+        item = Item().load(item_id, user=user)
+        description = item['description']
+
+        other_entity = ET.SubElement(dataset, 'otherEntity')
+        ET.SubElement(other_entity, 'entityName').text = item['name']
+        ET.SubElement(other_entity, 'entityDescription').text = description if\
```

Comment:
Similar comment here to the one I made for abstract: If the element is optional, I think omitting the entire element is better than saying "None"